react/src/components/PostHogFeature.tsx

```diff
@@ -0,0 +1,88 @@
+import { useFeatureFlagPayload, useFeatureFlagVariantKey, usePostHog } from '../hooks'
+import React, { useCallback, useEffect, useRef } from 'react'
+import { PostHog } from '../context'
+
+export type PostHogFeatureProps = React.HTMLProps<HTMLDivElement> & {
+    flag: string
+    children: React.ReactNode | ((payload: any) => React.ReactNode)
+    fallback: React.ReactNode
+    match?: string | boolean
+    visibilityObserverOptions?: IntersectionObserverInit
+}
+
+export function PostHogFeature({
+    flag,
+    match,
```

Comment:
Why do we need the implementer to pass in the match value? Can't we retrieve this ourselves if they've initialized a provider/client?

Reply:
This represents what you want to match on: when there are multiple variants for example, do you want to show this component for 'test', 'test-2' or 'control' ?

Reply:
Oh i see, it's not the actual value returned from decide